### DISCUSSION TITLE
Using DataLine as an inline tool with Espanso

### DISCUSSION THREAD

**Discussion by vasnt** [RamiAwar/dataline#245] [Ideas]
I prefer to see any AI tool as "Inline" i.e. https://espanso.org/ can autocomplete wherever i am in. 
if i am writing message in discord/irc chat or emailing someone or writing docs/notes etc. 

e.g. if i need to email data of last years' profit to my partner. 
I'll be writing an email like : 
my last year's profit was ......(to be filled by dataline upon calling it or DATALINE can be on auto-pilot mode)...

**Comment by RamiAwar** [comment]
It's possible to build something like this by adding an additional query route to dataline that pre-specifies a certain connection for example. That's the only real variable that goes into a query.

In that scenario we also wouldn't care about message history I'm guessing. Kind of a chat-autocomplete feature.

But a lot of the features wouldn't be available (charts for example) and you'd only want the data results maybe. Idk, seems pretty specific and custom, and hard to work with. It's much easier to just go to the DataLine tab, have a conversation, copy the data you want. I don't think many people have answers that are so straightforward.

That being said, not against something like this if we narrow it down and it makes sense.

**Reply by vasnt** [reply]
- dataline Along with espanso is just one flavour. 
- Getting data using dataline independently (like autocomplete) can be main feature. 
- personally, i feel this as cumbersome step: "going to DataLine tab, have a conversation, copy the data you want." i think that is why espanso is so popular.

**Reply by RamiAwar** [reply]
rightt but this one works with basic stuff is my point. maybe you can answer some questions like this, but realistically, would anyone use this often enough to want espanso autocomplete to query their data? that's the part I'm doubting.

**Reply by vasnt** [reply]
people dont use espanso because it may not be AI based or everyone has limited memory capacity so one can recall only few variables. 
While dataline/any other AI tool can answer from any given question so memorising variables is not an issue. 
On the other hand: raycast kind of program is quite popular as it is quick launcher. & this topic is like quick (data) fetcher which would be definitely useful to many users. 
People also use windows program named as "Everything" for quickly finding files. So i belive any-thing which is quick and/or searches/finds/fetches data is useful.

**Reply by RamiAwar** [reply]
Okay, I'm not very convinced but that doesn't matter. We can keep this open as a feature idea and let people vote on it! It's not really aligned with any of the directions we're going in at the moment so won't work on it anytime soon unless there's significant demand.
